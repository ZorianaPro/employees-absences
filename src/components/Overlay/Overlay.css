.Overlay {
  display: block;
  opacity: 1;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 7;
  overflow: scroll;
}

.Overlay svg {
  height: 24px;
  width: 24px;
}

.Overlay .Overlay-Mask {
  background-color: rgba(59, 65, 74, 0.5);
  display: block;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
}

.Overlay .Overlay-Container {
  display: flex;
  position: relative;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  z-index: 1;
  width: 100%;
  height: 100%;
}

.Overlay .Overlay-Content {
  position: absolute;
  top: 30%;
  left: 50%;
  background-color: var(--colorWhite);
  width: 30em;
  transform: translate(-50%, -30%);
  padding: 1.5em;
  border-radius: 0.8em;
}

.Overlay .Overlay-Close {
  position: absolute;
  right: 2em;
  top: 2em;
  cursor: pointer;
  font-size: var(--fontSize16);
  z-index: 11;
}

.Overlay .Overlay-Content > * {
  margin: auto;
  z-index: 1;
}