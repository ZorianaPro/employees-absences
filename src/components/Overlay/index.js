import React from 'react';
import XCloseSVG from '../XCloseSVG';
import './Overlay.css';

const Overlay = ({
  children,
  onClose
}) => (
  <div className="Overlay">
    <div className="Overlay-Mask"/>
    <div className="Overlay-Container">
      <div className="Overlay-Content">
        <div className="Overlay-Close"
          onClick={
            onClose
          }
        >
          <XCloseSVG/>
        </div>
        {
          children
        }
      </div>
    </div>
  </div>
);

export default Overlay;