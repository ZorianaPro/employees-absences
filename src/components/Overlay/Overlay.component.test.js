import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Overlay from './';

describe('Overlay', () => {
  let overlay;

  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Overlay />))
      .not
      .toThrow();
  });

  it('renders as expected', () => {
    expect(shallow(<Overlay />))
      .toMatchSnapshot();
  });

  describe('children', () => {
    it('renders all the children', () => {
      expect(
        shallow(
          <Overlay>
            <p>test</p>
          </Overlay>
        ).exists('p')
      ).toBe(true);
    });
  });

  describe('onClose', () => {
    const onClose = jest.fn();
    beforeEach(() => {
      overlay = mount(
        <Overlay
          onClose={ onClose }
        />);
      act(() => {
        overlay
          .find('.Overlay-Close')
          .simulate('click');
      });
    });

    it('calls onClose callback', () => {
      expect(onClose)
        .toHaveBeenCalledTimes(1);
    });
  });
});