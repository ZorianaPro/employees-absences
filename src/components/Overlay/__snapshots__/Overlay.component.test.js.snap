// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay renders as expected 1`] = `
<div
  className="Overlay"
>
  <div
    className="Overlay-Mask"
  />
  <div
    className="Overlay-Container"
  >
    <div
      className="Overlay-Content"
    >
      <div
        className="Overlay-Close"
      >
        <XCloseSVG />
      </div>
    </div>
  </div>
</div>
`;
